Classes/Service/Shopware/CustomerClient.php

```diff
@@ -0,0 +1,132 @@
+ *
+ * @package Portrino\PxShopware\Service\Shopware
+ */
+class CustomerClient extends AbstractShopwareApiClient implements CustomerClientInterface
+{
+
+    /**
+     * @param string $term
+     * @param int $limit
+     * @param bool $doCacheRequest
+     * @param array $params
+     * @return ObjectStorage
+     */
+    public function findByTerm($term, $limit = -1, $doCacheRequest = true, $params = [])
+    {
+        $shopwareModels = new ObjectStorage();
+
+        /**
+         * only search for id if term is integer
+         */
+        if (is_numeric($term)) {
+            $filter = [
+                [
+                    'property' => 'id',
+                    'expression' => '=',
+                    'value' => $term
+                ],
+                [
+                    'operator' => '1',
+                    'property' => 'number',
+                    'expression' => 'LIKE',
+                    'value' => '%' . $term . '%'
+                ],
+            ];
+        } else {
+            $filter = [
+                [
+                    'property' => 'email',
+                    'expression' => 'LIKE',
+                    'value' => '%' . $term . '%'
+                ],
+                [
+                    'operator' => 'OR',
+                    'property' => 'firstname',
+                    'expression' => 'LIKE',
+                    'value' => '%' . $term . '%'
+                ],
+                [
+                    'operator' => 'OR',
+                    'property' => 'lastname',
+                    'expression' => 'LIKE',
+                    'value' => '%' . $term . '%'
+                ]
+            ];
+        }
+
+        ArrayUtility::mergeRecursiveWithOverrule($params, [
+            'limit' => $limit,
+            'sort' => [
+                [
+                    'property' => 'number',
+                    'direction' => 'DESC'
+                ]
+            ],
+            'filter' => $filter
+        ]);
+
+        $result = $this->get($this->getValidEndpoint(), $params, $doCacheRequest);
+        if ($result) {
+            $token = (isset($result->pxShopwareTypo3Token)) ? (bool)$result->pxShopwareTypo3Token : false;
+            if (isset($result->data) && is_array($result->data)) {
+                foreach ($result->data as $data) {
+                    if (isset($data->id)) {
+                        /** @var \Portrino\PxShopware\Domain\Model\AbstractShopwareModel $shopwareModel */
+                        $shopwareModel = $this->objectManager->get($this->getEntityClassName(), $data, $token);
+                        if ($shopwareModel != null) {
+                            $shopwareModels->attach($shopwareModel);
+                        }
+                    }
+                }
+            }
+        }
+        return $shopwareModels;
+    }
+
+    /**
+     * @return string
+     */
+    public function getEndpoint()
+    {
+        return self::ENDPOINT;
```

Comment:
I think it is better to use the interface name instead of self. At some point of time you will remove the Interface and then this will maybe fail ;-) just cosmetic stuff

Reply:
I also think that self is not inheritance aware... 

https://stackoverflow.com/questions/13613594/overriding-class-constants-vs-properties

Reply:
Well, right now it's the same for every Shopware client class. But I can change it to the interface instead of self. I think static instead of self makes no sense here...